libparsec/crates/client/src/certif/store.rs

```diff
@@ -33,6 +33,42 @@ use crate::certif::CertifPollServerError;
 
 use super::{realm_keys_bundle::RealmKeys, CertificateOps, InvalidCertificateError};
 
+/// This enum list all possible states for a realm.
+///
+/// Bootstrapping a user realm just correspond to uploading its initial realm role
+/// manifest.
+/// Bootstrapping a workspace realm (the famous "workspace bootstrap") requires two
+/// additional certificates to be upload in a non-atomic way, hence the need to know
+/// at what state we currently are in order to be able to achieve such bootstrap in
+/// an idempotent way.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum RealmBootstrapState {
+    /// Step 0
+    ///
+    /// The realm has been created on our device but doesn't exist yet on the server
+    /// (i.e. we haven't send the initial realm role certificate).
+    LocalOnly,
+    /// Step 1
+    ///
+    /// The realm exists on the server (i.e. initial realm role certificate has been uploaded).
+    /// Notes:
+    /// - The user realm doesn't go beyond this state.
+    /// - For a workspace realm, this is the first step of the workspace bootstrap.
+    CreatedInServer,
+    /// Step 2
```

Comment:
```suggestion
    /// Step 2: for workspace only
```

Reply:
And maybe we could add that to next step's doc for consistency, but the name WorkspaceBootstapped is clear enough

Reply:
:+1: done.